### PR TITLE
Add logic of autoindex

### LINKED ISSUE #19467: [Feature]:  Add autoIndex feature
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe.

The server is configured with default index build parameters.
Different recall precisions can be specified by `level` parameter when searching.
The default value of the level parameter is 1, which means the lowest level among multiple levels.
The client only needs to specify the metric_type when creating an index or searching, and optionally specify the `level` parameter to determine the different levels of recall when searching.

### Describe the solution you'd like.

_No response_

### Describe an alternate solution.

_No response_

### Anything else? (Additional Context)

_No response_

### PR DESCRIPTION
**Pull request by czs007**
fix #19467 
Signed-off-by: zhenshan.cao <zhenshan.cao@zilliz.com>

### CODE REVIEW

File: internal/metastore/db/dbmodel/index.go [hunk 1]
```diff
@@ -14,18 +14,20 @@ import (
...
+	IndexID         int64     `gorm:"index_id"`
+	IndexName       string    `gorm:"index_name"`
+	IndexParams     string    `gorm:"index_params"`
+	TypeParams      string    `gorm:"type_params"`
+	CreateTime      uint64    `gorm:"create_time"`
+	IsDeleted       bool      `gorm:"is_deleted"`
+	CreatedAt       time.Time `gorm:"created_at"`
+	UpdatedAt       time.Time `gorm:"updated_at"`
+	IsAutoIndex     bool      `gorm:"is_auto_index"`
+	UserIndexParams string    `gorm:"index_params"`
```
Comment (czs007) [inline internal/metastore/db/dbmodel/index.go:30]: wrong gorm

File: tests/python_client/testcases/test_connection.py [hunk 2]
```diff
@@ -542,7 +542,7 @@ def test_connection_connect_params(self, host, port, connect_name):
                                                  check_items={ct.dict_content: {"address": f"{host}:{port}",
                                                                                 "user": ""}})
 
-    @pytest.mark.tags(ct.CaseLabel.L1)
+    @pytest.mark.tags(ct.CaseLabel.L2)
```
Comment (jennyli-z) [inline tests/python_client/testcases/test_connection.py:545]: Curious about why change the test case Level?

File: internal/metastore/db/dbmodel/index.go (hunk 1 above)
Comment (czs007) [reply internal/metastore/db/dbmodel/index.go:30]: meta.sql should also add corresponding field.

File: tests/python_client/testcases/test_connection.py (hunk 2 above)
Comment (czs007) [reply tests/python_client/testcases/test_connection.py:545]: @yanliang567  aggreed. 
You CAN NOT pass  this case in your local machine！！
I usually run the L0 L1 tests locally before submitting the pr. 
so...

File: internal/util/autoindex/function.go [hunk 3]
```diff
@@ -0,0 +1,132 @@
+// Licensed to the LF AI & Data foundation under one
```
Comment (xiaofan-luan) [inline internal/util/autoindex/function.go:1]: check if this library will work, should be much easier to config
https://github.com/yidane/formula

Comment (xiaofan-luan) [reply internal/util/autoindex/function.go:1]: or might this library https://github.com/Sandertv/go-formula

File: internal/proxy/task_index.go [hunk 4]
```diff
@@ -0,0 +1,678 @@
...
+				indexParamsMap[k] = v
+			}
+		} else {
+			indexParamsMap[kv.Key] = kv.Value
+		}
+	}
+
+	if isVecIndex {
+		specifyIndexType, exist := indexParamsMap[common.IndexTypeKey]
+		if Params.AutoIndexConfig.Enable {
```
Comment (xiaofan-luan) [inline internal/proxy/task_index.go:132]: if Params.AutoIndexConfig.Enable not true, where did we check index type and fill in default index?

File: internal/proxy/task_index.go [hunk 5]
```diff
@@ -0,0 +1,678 @@
...
+	if err != nil { // err is not nil if collection not exists
+		return err
+	}
+	gibpt.collectionID = collectionID
+
+	if gibpt.IndexName == "" {
+		gibpt.IndexName = Params.CommonCfg.DefaultIndexName
+	}
+
+	resp, err := gibpt.indexCoord.GetIndexBuildProgress(ctx, &indexpb.GetIndexBuildProgressRequest{
```
Comment (xiaofan-luan) [inline internal/proxy/task_index.go:567]: change to describe index?

File: internal/proxy/task_search.go [hunk 6]
```diff
@@ -89,8 +96,50 @@ func getPartitionIDs(ctx context.Context, collectionName string, partitionNames
 	return partitionIDs, nil
 }
 
-// parseQueryInfo returns QueryInfo and offset
-func parseQueryInfo(searchParamsPair []*commonpb.KeyValuePair) (*planpb.QueryInfo, int64, error) {
+func parseSearchParams(searchParamsPair []*commonpb.KeyValuePair) (string, error) {
+	searchParamStr, err := funcutil.GetAttrByKeyFromRepeatedKV(SearchParamsKey, searchParamsPair)
+	if Params.AutoIndexConfig.Enable {
```
[the remark this answers is not in the document]
Comment (czs007) [reply internal/proxy/task_search.go:101]: For now, we wil overwrite old parameters, and extraneous  parameters are ignored by the underlying engine.

**Pull request merged**